com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -862,7 +862,7 @@ public override void OnNetworkSpawn()
             {
                 TryCommitTransformToServer(m_Transform, m_CachedNetworkManager.LocalTime.Time);
             }
-            m_LocalAuthoritativeNetworkState = m_ReplicatedNetworkState.Value;
+            m_LastSentState = m_LocalAuthoritativeNetworkState = m_ReplicatedNetworkState.Value;
```

Comment:
I will look into this issue in #2102 since there are some adjustments that could fix the issue you might be experiencing with client-side authority.  m_LastSentState should be only set by the authority as that is all it is used for, and this change will set it on both non-authoritative and authoritative (although setting it on the non-authoritative side shouldn't have any real impact to non-authoritative instances).